### ISSUE TITLE
Multiple SetMemberLengthMapping break dynamic-length lists

### ISSUE THREAD

**Issue by akielijan** [Xeeynamo/Xe.BinaryMapper#3]
Version: 1.4.1 (NuGet)

Hi, I tried to map multiple lists in the structure with different length indicators. I moved my `SetMemberLengthMapping` instructions to the static constructor of the class. 

Assume the following definitions:
``` csharp
public class SimpleStructure
    {
        [Data] public byte Value { get; set; }
    }
    public class SampleClass
    {
        static SampleClass()
        {
            BinaryMapping.SetMemberLengthMapping<SampleClass>(nameof(SampleClass.Structures), (o, m) => o.Length);
            BinaryMapping.SetMemberLengthMapping<SampleClass>(nameof(SampleClass.Structures2), (o, m) => o.Length2);
        }
        [Data]
        public int Length
        {
            get => Structures.TryGetCount();
            set => Structures = Structures.CreateOrResize(value);
        }
        [Data] public List<SimpleStructure> Structures { get; set; }

        [Data]
        public int Length2
        {
            get => Structures2.TryGetCount();
            set => Structures2 = Structures2.CreateOrResize(value);
        }
        [Data] public List<SimpleStructure> Structures2 { get; set; }
    }
```

When I'm using the one mapping (e.g. comment the Length2, Structures2 and second `SetMemberLengthMapping`), my tests seem to be fine with the following data (int 0):
``` csharp
byte[] hexData = new byte[] { 0x00, 0x00, 0x00, 0x00 };
```

Basically, it loads the int and drops the list creation/resize. But when I keep using two ints 0 (with uncommented length, list and mapping):
``` csharp
byte[] hexData = new byte[] { 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00 };
```
, it throws `EndOfStreamException` when trying to create a list `Structures2` and loads the SimpleStructure (even though it shouldn't).

Sorry for the class naming, it was just a PoC on my side. For the time being, I created a class that aggregates both the length (int) and the values (List<T>) - this way I limit the usage of `SetMemberLengthMapping` to one call.

I know you have a version 1.5, but it's breaking the API and I'm waiting for new documentation. The issue may still happen on newer version, as well ;)

**Comment by Xeeynamo** [comment]
Hello!
Sorry for coming back after 3 weeks. I acknowledged the bug and I just pushed a fix for it!

Also you are right about the version `1.5.0`: it was supposed to be pushed as a pre-release, but life changes forced me to think about it later and it has been left like that for a while. I am working on a 1.5.1 which will fix your issue and will add instructions on which API has been changed (or broken) and why. I am sorry if this caused you any trouble.
